kvdb-rocksdb/src/lib.rs

```diff
@@ -443,15 +443,24 @@ impl Database {
 							stats_total_bytes += key.len();
 							batch.delete_cf(cf, &key).map_err(other_io_err)?
 						}
-						DBOp::DeletePrefix { col: _, prefix } => {
+						DBOp::DeletePrefix { col, prefix } => {
+							let iter_prefix_delete = |prefix: &[u8], batch: &mut WriteBatch| -> io::Result<()> {
+								for (key, _) in self.iter_with_prefix(col, &prefix[..]) {
+									batch.delete_cf(cf, &key[..]).map_err(other_io_err)?;
+								}
+								Ok(())
+							};
```

Comment:
```suggestion
							let end_prefix = end_prefix(&prefix[..]);
							let end_range = end_prefix.unwrap_or_else(|| vec![u8::max_value(); 16]);
							batch.delete_cf(cf, &prefix[..], &end_range[..]).map_err(other_io_err)?;
							if end_prefix.is_none() {
								let prefix = if prefix.len() > end_range.len() { &prefix[..] } else { &end_range[..] };
								for (key, _) in self.iter_with_prefix(col, prefix) {
									batch.delete_cf(cf, &key[..]).map_err(other_io_err)?;
								}
							}
							Ok(())
```
I suggest the following code (and remove the code below). This will also optimize the case when e.g. we have several keys starting with `[255, 255, 255, 255]` and are calling `delete_prefix(&[255, 255])`, the iterator will be empty in this case, so it will issue on one range deletion.